src/main/java/com/google/sps/servlets/FindLocationServlet.java

```diff
@@ -13,13 +17,21 @@
   * Given the correlator, this will update the database to reflect that the user
   * clicked 'find nearby location'.
   *
-  * request - This includes a correlator of type String for the current user.
-  * response - An empty 200 response 
+  * @param request This includes a correlator for the current user.
+  * @param response A 200 status or 400 error.
   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     
-    //TODO - implement the rest of this function (@Kaleab)
+    final String correlator = RequestUtils.getCorrelator(request);
+
+    if (correlator.isEmpty()) {
```

Comment:
I noticed that you're making this correlator check in most of the Java files, I don't know how feasible it would be but is there any way you could make this into one helper method that you can access multiple times? It could help with readability in the long term

Reply:
I did think about putting it in the utilities file but i think the way its structured now( if not valid send 404, else set to true and send 200) makes more sense logically to me when reading it. could be wrong tho